src/main/grammars/RustParser.bnf

```diff
@@ -984,6 +984,37 @@ MacroDefinition ::= MacroRulesInvocation identifier ItemMacroArg {
   elementTypeFactory = "org.rust.lang.core.stubs.StubImplementationsKt.factory"
 }
 
+MacroDefinitionPatternList ::= <<macro_definition_delim (MacroDefinitionPattern ';'?)* >>
+
+private meta macro_definition_delim ::= '(' <<param>> ')' ';'
+                                      | '{' <<param>> '}'
+                                      | '[' <<param>> ']' ';' { pin = 1 }
+MacroDefinitionPattern ::= MacroPattern '=>' MacroBody
+MacroBinding ::= (identifier | "type")
```

Comment:
Hm, so inside a macro you are allowed to use keyword as an identifier? Weird. I think that perhaps just listing all the keywords here is the best solution.